### ISSUE TITLE
[2019/11/27] Machine Learning 輪講

### ISSUE THREAD

**Issue by agatan** [wantedly/machine-learning-round-table#27]
## Why

Machine Learning 輪講は最新の技術や論文を追うことで、エンジニアが「技術で解決できること」のレベルをあげていくことを目的にした会です。

prev. #24 

## What

話したいことがある人はここにコメントしましょう！
面白いものを見つけた時点でとりあえず話すという宣言だけでもしましょう！

**Comment by agatan** [comment]
## Faster Neural Networks Straight from JPEG

https://eng.uber.com/neural-networks-jpeg/

**Comment by agatan** [comment]
## [1911.09665] Adversarial Examples Improve Image Recognition

https://arxiv.org/abs/1911.09665

**Comment by agatan** [comment]
## Filter Response Normalization Layer: Eliminating Batch Dependence in the Training of Deep Neural Networks

https://arxiv.org/abs/1911.09737

**Comment by tan-z-tan** [comment]
### LEVERAGING GRAMMAR AND REINFORCEMENT LEARNING FOR NEURAL PROGRAM SYNTHESIS
https://openreview.net/pdf?id=H1Xw62kRZ
プログラムを生成する
INPUT, OUTPUT, PROGRAMの三組から、「INPUT, OUTPUTを与えたらPROGRAMを生成する変換モデル」を学習する。

#### 偉いところ
- Exactに一致するプログラムでなくても、入力ー出力が一致するならスコアを与える
    - そのために、Backpropagationではなく強化学習として解く
- 明らかにSyntaxエラーなものを枝刈りする

### NEURO-SYMBOLIC PROGRAM SYNTHESIS
https://arxiv.org/pdf/1611.01855.pdf
プログラムを木構造で表す。
- A → B Cの確率分布（ネットワーク）
- B C → A の確率分布（ネットワーク）

それぞれを学習して、生成する。

### GENERATIVE CODE MODELING WITH GRAPHS
https://arxiv.org/pdf/1805.08490.pdf
プログラムの生成は文章の生成より構造的なもので周辺のコンテキストをもっと入れ込みたい。
AST (Abstract Syntax Tree)の中で、child, parent, nextSib, NextUseなどの決められた関係性にあるノードの情報を次のTokenの生成に使う。関係性のあるノードからエッジが引かれるのでグラフ構造になる。

ASTのルートから初めて、現在のEmbedding **h** に対して関係性のあるノードの情報（それぞれのノードのEmbeddingを変換したものをElement-wiseに足したもの）を入力にして次のTokenを生成する。これを再帰的に行ってASTの各ノードがEmbedding **h_i**である木を作る。
各ノードのEmbeddingからProductionルールや変数を選ぶのは、e(**h_i**) -> rule みたいに一意に決まる関数を学習しておく。

**Comment by agatan** [comment]
##  Momentum Contrast for Unsupervised Visual Representation Learning

https://arxiv.org/abs/1911.05722
